test/functional/dynamic_forall/resource-segment/tests/test-dynamic-forall-resource-RangeSegment.hpp

```diff
@@ -66,7 +66,9 @@ TYPED_TEST_P(DynamicForallResourceRangeSegmentTest, RangeSegmentForallResource)
   using POLICY_LIST = typename camp::at<TypeParam, camp::num<2>>::type;
 
 
+#if defined(RAJA_DEVICE_ACTIVE)
```

Comment:
This means cuda or hip are active? Elsewhere we use another macro that also has DEVICE in it to say that we're in the device compile pass and this double meaning of DEVICE is confusing.

Reply:
I didn't think too hard about it. I was following the macro usage that already existed in the file. I added that guard to squash compiler warnings about unused variables. If you have a specific change in mind, or suggestion, please commit it or let me know what you want me to do. Thank you.

Reply:
Should we use `RAJA_DEVICE_IS_AVAILABLE`? Since that is what it means

Reply:
which is different that `RAJA_DEVICE_CODE` which means the code is specific to the device

Reply:
To me `RAJA_DEVICE_IS_AVAILABLE` is also confusing. Fundamentally its about what RAJA_DEVICE means, I don't think its clear that sometimes it means device code and other times it means a programming model.
Ya I commented on a couple things here that I didn't notice earlier, so they're not really about this PR as much as they are things I would otherwise forget about if I didn't say them now, maybe I should move this to an issue.
But, to be clear I don't necessarily think we should change this here.

Reply:
Okay, so we need something that says RAJA has been compiled to the GPU and the following block of code will run when the GPU is available, but is not kernel code.

Reply:
Moving discussion to issue #1395.